jupyter_collaboration/handlers.py

```diff
@@ -321,72 +261,51 @@ def on_close(self) -> None:
+        # Remove the room from the websocket server
         assert self.websocket_server is not None
-        file_format, file_type, file_path = self.get_file_info()
-        self.log.debug("Deleting Y document from memory: %s", file_path)
+        self.log.info("Deleting Y document from memory: %s", self.room.room_id)
         self.websocket_server.delete_room(room=self.room)
 
-    def on_document_change(self, target, event):
-        if target == "state" and "dirty" in event.keys:
-            dirty = event.keys["dirty"]["newValue"]
-            if not dirty:
-                # we cleared the dirty flag, nothing to save
-                return
-        if self.saving_document is not None and not self.saving_document.done():
-            # the document is being saved, cancel that
-            self.saving_document.cancel()
-            self.saving_document = None
-        self.saving_document = asyncio.create_task(self.maybe_save_document())
-
-    async def maybe_save_document(self):
-        assert isinstance(self.room, DocumentRoom)
-        seconds = self.settings["collaborative_document_save_delay"]
-        if seconds is None:
-            return
-        # save after X seconds of inactivity
-        await asyncio.sleep(seconds)
-        # if the room cannot be found, don't save
-        try:
-            file_format, file_type, file_path = self.get_file_info()
-        except Exception:
-            return
-        self.log.debug("Opening Y document from disk: %s", file_path)
-        async with self.lock:
-            model = await ensure_async(
-                self.contents_manager.get(file_path, type=file_type, format=file_format)
-            )
-        if self.last_modified < model["last_modified"]:
-            # file changed on disk, let's revert
-            self.log.debug("Reverting file that had out-of-band changes: %s", file_path)
-            self.room.document.source = model["content"]
-            self.last_modified = model["last_modified"]
-            return
-        if model["content"] != self.room.document.source:
-            # don't save if not needed
-            # this also prevents the dirty flag from bouncing between windows of
-            # the same document opened as different types (e.g. notebook/text editor)
-            model["format"] = file_format
-            model["content"] = self.room.document.source
-            self.log.debug("Saving Y document to disk: %s", file_path)
-            async with self.lock:
-                model = await ensure_async(self.contents_manager.save(model, file_path))
-                self.last_modified = model["last_modified"]
-        self.room.document.dirty = False
+        # Clean room
+        del self.room
+        self.log.info("Room %s deleted", self._room_id)
+
+        # Clean the file loader if there are not rooms using it
+        _, _, file_id = decode_file_path(self._room_id)
+        file = self.files[file_id]
+        if file.number_of_subscriptions() == 0:
```

Comment:
As you use a name, it makes more sense to define that as a `property` rather than a `method`.